pkg/checklist/service.go

```diff
@@ -52,19 +52,17 @@ func (s *service) ToggleDone(ctx context.Context, id int64) error {
 	}
 
 	task.Done = !task.Done
-	err = s.repository.Update(ctx, task)
-	if err != nil {
+	if err = s.repository.Update(ctx, task); err != nil {
 		return fmt.Errorf("could not toggle task: %v", err)
 	}
 	return nil
 }
 
 func (s *service) Remove(ctx context.Context, id int64) error {
-	err := s.repository.DeleteByID(ctx, id)
-	if err == todo.ErrTaskNotFound {
-		return err
-	}
-	if err != nil {
+	if err := s.repository.DeleteByID(ctx, id); err != nil {
+		if err == todo.ErrTaskNotFound {
+			return err
+		}
```

Comment:
Increased indentation in the code is an indication that it could be written better. It's usually better to optimize for as little indentation is possible. Makes the code easier to follow linearly - top to bottom.

Reply:
https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow